src/pybind/mgr/dashboard/controllers/__init__.py

```diff
@@ -662,14 +667,23 @@ def inner(*args, **kwargs):
             params = get_request_body_params(cherrypy.request)
             kwargs.update(params)
 
-            ret = func(*args, **kwargs)
+            req_version = '15'
+            if 'vnd.ceph.api.v' in cherrypy.request.headers.get('Content-Type'):
+                req_version = re.match(r"\d\d", cherrypy.request.headers['Content-Type'])
+                if req_version not in ['15']:
+                    req_version = '15'
+
+            if req_version == version:
+                ret = func(*args, **kwargs)
+                # ret = func(*args, **kwargs)
+            # ret = func(*args, **kwargs)
             if isinstance(ret, bytes):
                 ret = ret.decode('utf-8')
             if xml:
                 cherrypy.response.headers['Content-Type'] = 'application/xml'
                 return ret.encode('utf8')
             if json_response:
-                cherrypy.response.headers['Content-Type'] = 'application/json'
+                cherrypy.response.headers['Content-Type'] = 'application/vnd.ceph.api.v{}+json'.format(version)
```

Comment:
If we enable `json_out` in the same way as we did for `json_in` (dashboard/module.py), we could configure that module to add our stuff:

```diff
# module.py
+ def our_handler(*args, **kwargs):
+   ...

  ...
  'tools.json_in.on': True,                                                                
  'tools.json_in.force': False,
+ 'tools.json_in.on': True,                                                                
+ 'tools.json_out.handler': our_handler,
+ 'tools.json_out.content_type': None # this is needed if we want to set-up the content-type and avoid it from being ovewritten
```